src/DynamoCoreWpf/ViewModels/Menu/PreferencesViewModel.cs

```diff
@@ -1256,7 +1256,7 @@ internal void HideUnhideNamespace(bool hide, string library, string namespc)
         private void AddPythonEnginesOptions()
         {
             var options = new ObservableCollection<string> { Res.DefaultPythonEngineNone };
-            foreach (var item in PythonEngineManager.Instance.AvailableEngines)
+            foreach (var item in PythonEngineManager.Instance.AvailableEngines.GroupBy(x=>x.Name).FirstOrDefault().ToList())
```

Comment:
So is it ok to have Dynamo expect duplicate engine names and handle them accordingly ?

Reply:
because `AvailableEngines` is a public observable field theres not much we can do to stop a client from adding another engine name. I think ideally this would not be public and we'd decouple access with an API. I don't really want to change this at this point in the release cycle. I can file a followup to fix this and mark this field obsolete. Then we can remove this logic and rethink what should happen in the case of duplicate engine names.

Reply:
Why did I make AvailableEngines public ?...
Could you mark it as Obsolete and say that it might go away some time in the future ? that should have a minimal effect

Reply:
If for some reason that (mark it as Obsolete) seems risky, then I am ok with postponing it
I agree with the rest of what you said

Reply:
sure, I will make it obsolete.